### PR TITLE
[184] Implement show endpoint for api/v2/contacts

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/X1EkaWrR/184-add-api-v2-contacts-id-show-endpoint

### Changes proposed in this pull request

- Add a `api/v2/contacts/:id` endpoint

### Guidance to review

Run the following locally:

```
curl --location --request GET 'localhost:3001/api/v2/contacts/:id' # Pass the contact id and a bearer token
```

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Add /api/v2/contacts/<id> show endpoint](https://trello.com/c/X1EkaWrR/184-add-api-v2-contacts-id-show-endpoint)

**Review by gpeng — APPROVED** [review]

File: spec/requests/api/v2/providers/contacts_spec.rb [hunk 1]
```diff
@@ -1,6 +1,27 @@
 require "rails_helper"
 
-RSpec.describe "/api/v2/providers/<accredited_body_code>/contacts", type: :request do
```
Comment (misaka) [inline spec/requests/api/v2/providers/contacts_spec.rb:3]: Should this file be moved to `spec/requests/api/v2/contacts_spec.rb`?

Also, on a related note, if we ever add an endpoint for listing contacts we'll probably want it to be scoped to the provider. But I guess that's a can we can kick down the road for now.

**Review by misaka — COMMENTED** [review]
Think we might want to move the spec file, but otherwise all good.

File: spec/requests/api/v2/providers/contacts_spec.rb (hunk 1 above)
Comment (darokel) [reply spec/requests/api/v2/providers/contacts_spec.rb:3]: done https://github.com/DFE-Digital/teacher-training-api/pull/1585/commits/594b4fef2b189ce4e88050843f22bfd93f6a13d6

**Review by misaka — APPROVED** [review]

**Comment by aliuk2012** [comment]
@darokel this will need to be rebased off master to get the fixes for the failing spec

**Comment by darokel** [comment]
> @darokel this will need to be rebased off master to get the fixes for the failing spec

Thanks mate, saved me some time!

**Pull request merged**